src/main/java/seedu/address/model/task/DateTime.java

```diff
@@ -0,0 +1,75 @@
+package seedu.address.model.task;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Task's dateTime in PlaNUS.
+ * Guarantees: immutable; is valid as declared in {@link #isValidDateTime(String)}
+ */
+public class DateTime {
+
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "DateTime should be in the format of dd-MM-yyyy HH:mm.";
+    public static final String VALIDATION_REGEX =
+            "^(3[01]|[12][0-9]|0[1-9])-(1[0-2]|0[1-9])-[0-9]{4} (2[0-3]|[01]?[0-9]):([0-5]?[0-9])$";
+
+    private static final String DEFAULT_DATETIME = "";
+
+    public final String value;
+
+
+    /**
+     * Constructs a {@code DateTime}.
+     *
+     * @param dateTime A valid date and time.
+     */
+    public DateTime(String dateTime) {
+        requireNonNull(dateTime);
+        checkArgument(isValidDateTime(dateTime), MESSAGE_CONSTRAINTS);
+        value = dateTime;
+    }
+
+    /**
+     * Constructs a default {@code DateTime}.
+     *
+     * Caveat: Only called by defaultDateTime method.
+     */
+    private DateTime() {
+        value = DEFAULT_DATETIME;
+    }
+
+    /**
+     * Returns true if a given string is a valid phone number.
+     */
+    public static boolean isValidDateTime(String test) {
+        return test.matches(VALIDATION_REGEX);
+    }
+
+    /**
+     * Constructs an empty Phone when user didn't provide the phone field.
```

Comment:
Should be "constructs an empty DateTime when user does not provide the dateTime field